x/crisis/handler.go

```diff
@@ -0,0 +1,56 @@
+package crisis
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const ModuleName = "crisis"
+
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+
+		switch msg := msg.(type) {
+		case MsgVerifyInvariance:
+			return handleMsgVerifyInvariance(ctx, msg, k)
+		default:
+			return sdk.ErrTxDecode("invalid message parse in crisis module").Result()
+		}
+	}
+}
+
+func handleMsgVerifyInvariance(ctx sdk.Context, msg MsgVerifyInvariance, k Keeper) sdk.Result {
+
```

Comment:
we need to charge a large fee here to prevent spam

Reply:
do we really need an extra fee on top of the gas cost?

Reply:
I think so - see https://github.com/cosmos/cosmos-sdk/pull/3656#discussion_r262044275